### ISSUE TITLE
Testing

### ISSUE THREAD

**Issue by enochtsang** [UCSolarCarTeam/Solar-Simulation#5]
Suggestions on how to blackbox and whitebox test?

**Comment by sneakyturtle** [comment]
What is this blackbox and whitebox test?

**Comment by enochtsang** [comment]
Blackbox is like overall testing, one example would be starting up the program and clicking buttons and expecting a certain result.

Whitebox testing is like unit testing, where you run tests on specific functions and expect a certain return or exception. 

This is something we need to know in 511 and seng301 haha, good for you to learn now.

**Comment by sneakyturtle** [comment]
I know what it means. Do you just want us to do something like that for solar car?

**Comment by enochtsang** [comment]
Absolutely, so I'm looking for suggestions on how we should set up either testing. There is a unity testing framework but it looks like it's for Unity 4, not Unity 5. Can you take a closer look at that henry?

**Comment by sneakyturtle** [comment]
Sure

**Comment by sneakyturtle** [comment]
I found something free http://blogs.unity3d.com/2013/12/18/unity-test-tools-released/ 
It does unit testing, integration testing and also has an assertion component.
